sage_contact/repository/queryset/contact.py

```diff
@@ -0,0 +1,113 @@
+from django.db import models
+from django.db.models import QuerySet
+
+
+class LabelQuerySet(QuerySet):
+    """
+    Custom QuerySet for the Label model.
+    """
+
+    def search_by_name(self, name: str) -> QuerySet:
```

Comment:
use better algorithm for searching. You can get idea from `django-sage-blog` package.